### DISCUSSION TITLE
Trying to build opentelemetry-cpp for mac-os using bazle

### DISCUSSION THREAD

**Discussion by vinayakadj** [open-telemetry/opentelemetry-cpp#1646] [Q&A]
Hi I am trying to build opentelemetry-cpp using the command bazel build //... --host_javabase=@local_jdk//:jdk but I am getting an error :

INFO: From Linking external/com_github_opentelemetry_proto/libtrace_service_proto.so:
ld: warning: -undefined dynamic_lookup may not work with chained fixups
INFO: From Linking external/com_google_protobuf/libprotobuf.so:
ld: warning: -undefined dynamic_lookup may not work with chained fixups
INFO: From Linking external/com_google_protobuf/libprotobuf_lite.so:
ld: warning: -undefined dynamic_lookup may not work with chained fixups
INFO: From Linking external/com_github_opentelemetry_proto/libmetrics_service_proto.so:
ld: warning: -undefined dynamic_lookup may not work with chained fixups
ERROR: /Users/vinayaka.dj/Documents/AXP_code/opentelemetry_code_2/opentelemetry-cpp/ext/src/http/client/nosend/BUILD:3:11: Compiling ext/src/http/client/nosend/http_client_factory_nosend.cc failed: (Aborted): wrapped_clang failed: error executing command external/local_config_cc/wrapped_clang '-D_FORTIFY_SOURCE=1' -fstack-protector -fcolor-diagnostics -Wall -Wthread-safety -Wself-assign -fno-omit-frame-pointer -O0 -DDEBUG '-std=c++11' ... (remaining 50 argument(s) skipped)

Use --sandbox_debug to see verbose messages from the sandbox
ext/src/http/client/nosend/http_client_factory_nosend.cc:12:27: error: no member named 'nosend' in namespace 'opentelemetry::ext::http::client'; did you mean 'nostd'?
  return std::make_shared<http_client::nosend::HttpClient>();
                          ^~~~~~~~~~~~~~~~~~~
                          nostd
bazel-out/darwin_arm64-fastbuild/bin/api/_virtual_includes/api/opentelemetry/nostd/string_view.h:19:11: note: 'nostd' declared here
namespace nostd
          ^
ext/src/http/client/nosend/http_client_factory_nosend.cc:12:48: error: no member named 'HttpClient' in namespace 'opentelemetry::nostd'
  return std::make_shared<http_client::nosend::HttpClient>();
                          ~~~~~~~~~~~~~~~~~~~~~^
2 errors generated.
Error in child process '/usr/bin/xcrun'. 1
INFO: Elapsed time: 70.272s, Critical Path: 8.21s
INFO: 1041 processes: 442 internal, 599 darwin-sandbox.
FAILED: Build did NOT complete successfully
vinayaka.dj@banpc0wj2sz opentelemetry-cpp % 

Can some one please help me with this?

**Comment by owent** [comment]
Please add `--copt -DENABLE_TEST` after `bazel build` .
